bin/node/runtime/src/impls.rs

```diff
@@ -281,6 +282,98 @@ impl liberland_traits::OnLLMPoliticsUnlock<AccountId32> for OnLLMPoliticsUnlock
 	}
 }
 
+#[derive(
+	Clone,
+	Copy,
+	Eq,
+	PartialEq,
+	Encode,
+	Decode,
+	RuntimeDebug,
+	MaxEncodedLen,
+	scale_info::TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Coords {
+	lat: u64,
+	long: u64,
+}
+
+#[derive(
+	Clone,
+	Encode,
+	Decode,
+	RuntimeDebug,
+	MaxEncodedLen,
+	scale_info::TypeInfo,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct Metadata<MaxCoords: Get<u32>, MaxString: Get<u32>> {
+	demarcation: BoundedVec<Coords, MaxCoords>,
+	r#type: BoundedVec<u8, MaxString>,
+	status: BoundedVec<u8, MaxString>,
+}
+
+pub struct LandMetadataValidator;
+
+impl<StringLimit> pallet_nfts::traits::MetadataValidator<u32, u32, StringLimit>
+	for LandMetadataValidator
+{
+	fn validate_metadata(collection: u32, _: u32, data: &BoundedVec<u8, StringLimit>) -> bool {
+		// https://www.geeksforgeeks.org/check-if-two-given-line-segments-intersect/
+		fn intersection(a: Coords, b: Coords, c: Coords, d: Coords) -> bool {
+			fn ccw(a: Coords, b: Coords, c: Coords) -> bool {
+				(b.long - a.long) * (c.lat - b.lat) > (b.lat - a.lat) * (c.long - b.long)
+			}
+			ccw(a,c,d) != ccw(b,c,d) && ccw(a,b,c) != ccw(a,b,d)
+		}
+
+		if collection != 1 { return true; }
```

Comment:
Hmmm, 1 or 0, i guess this logic can be used both for land, and metavarse land....